backend/eventloop/repocred_reconciler.go

```diff
@@ -0,0 +1,156 @@
+package eventloop
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/go-logr/logr"
+	managedgitopsv1alpha1 "github.com/redhat-appstudio/managed-gitops/backend-shared/apis/managed-gitops/v1alpha1"
+	db "github.com/redhat-appstudio/managed-gitops/backend-shared/db"
+	sharedutil "github.com/redhat-appstudio/managed-gitops/backend-shared/util"
+	sharedresourceloop "github.com/redhat-appstudio/managed-gitops/backend/eventloop/shared_resource_loop"
+)
+
+const (
+	repoCredRowBatchSize            = 100              // Number of rows needs to be fetched in each batch.
+	repocredReconcilerInterval      = 10 * time.Minute // Interval in Minutes to reconcile Repository Credentials.
+	repoCredSleepIntervalsOfBatches = 1 * time.Second  // Interval in Millisecond between each batch.
+)
+
+// RepoCredReconciler reconciles RepositoryCredential entries
+type RepoCredReconciler struct {
+	client.Client
+	DB               db.DatabaseQueries
+	K8sClientFactory sharedresourceloop.SRLK8sClientFactory
+}
+
+// This function iterates through each entry of RepositoryCredential table in DB and updates the status of the CR.
+func (r *RepoCredReconciler) StartRepoCredReconciler() {
+	r.startTimerForNextCycle()
+}
+
+func (r *RepoCredReconciler) startTimerForNextCycle() {
+	go func() {
+		// Timer to trigger Reconciler
+		timer := time.NewTimer(time.Duration(repocredReconcilerInterval))
+		<-timer.C
+
+		ctx := context.Background()
+		log := log.FromContext(ctx).WithValues("component", "repocred-reconciler")
+
+		_, _ = sharedutil.CatchPanic(func() error {
+
+			// Reconcile RepositoryCredentials here
+			reconcileRepositoryCredentials(ctx, r.DB, r.Client, r.K8sClientFactory, log)
+
+			return nil
+		})
+
+		// Kick off the timer again, once the old task runs.
+		// This ensures that at least 'repocredReconcilerInterval' time elapses from the end of one run to the beginning of another.
+		r.startTimerForNextCycle()
+	}()
+
+}
+
+// /////////////
+// Reconcile logic for API CR To Database Mapping table and utility functions.
+// This will reconcile repository credential entries from ACTDM table and RepoistoryCredential table
+// /////////////
+func reconcileRepositoryCredentials(ctx context.Context, dbQueries db.DatabaseQueries, client client.Client, k8sClientFactory sharedresourceloop.SRLK8sClientFactory, log logr.Logger) {
```

Comment:
The unit tests appear not to cover `reconcileRepositoryCredentials`, we should add a unit tests for this